Jenkinsfile

```diff
@@ -0,0 +1,153 @@
+pipeline {
+    agent {
+        label 'python'
+    }
+
+    environment {
+        docker_repo = "indigodatacloud/app"
+    }
+
+    stages {
+        stage('Style Analysis') {
+            steps {
+                checkout scm
+                echo 'Running flake8..'
+                timeout(time: 5, unit: 'MINUTES') {
+                    sh 'tox -e pep8'
+                    echo 'Parsing pep8 logs..'
+                    step([$class: 'WarningsPublisher',
+                        parserConfigurations: [[
+                            parserName: 'Pep8', pattern: '.tox/pep8/log/*.log'
+                        ]], unstableTotalAll: '0', usePreviousBuildAsReference: true
+                    ])
+                }
+            }
+        }
+
+        stage('Unit tests') {
+            steps {
+                checkout scm
+                echo 'Computing unit testing coverage..'
+                sh 'tox -e cover'
+
+                echo 'Generating HTML report..'
+                publishHTML([allowMissing: false,
+                             alwaysLinkToLastBuild: false,
+                             keepAll: false,
+                             reportDir: 'cover',
+                             reportFiles: 'index.html',
+                             reportName: 'Coverage report',
+                             reportTitles: ''])
+
+                echo 'Generating Cobertura report..'
+                writeFile file: 'tox.ini.cobertura', text: '''[tox]
```

Comment:
This feels like it should be in a script in scm, and executed in the pipeline. It breaks the flow of the pipeline syntax here.

Reply:
+1 could be merged into tox.ini file so a coverage report can be generated